### PR TITLE
BUG: drop_duplicates raising for boolean arrow dtype with missing values

### LINKED ISSUE #54667: BUG: drop_duplicates raises unexpected error for bool[pyarrow] with null value
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
>>> dd = pd.DataFrame({"A": pd.Series([True, False, None], dtype="bool[pyarrow]")})
>>> dd.drop_duplicates()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "E:\pd_dev_0814\Lib\site-packages\pandas\core\frame.py", line 6568, in drop_duplicates
    result = self[-self.duplicated(subset, keep=keep)]
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\pd_dev_0814\Lib\site-packages\pandas\core\frame.py", line 6704, in duplicated
    result = self[subset[0]].duplicated(keep)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\pd_dev_0814\Lib\site-packages\pandas\core\series.py", line 2484, in duplicated
    res = self._duplicated(keep=keep)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\pd_dev_0814\Lib\site-packages\pandas\core\base.py", line 1368, in _duplicated
    return algorithms.duplicated(self._values, keep=keep)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "E:\pd_dev_0814\Lib\site-packages\pandas\core\algorithms.py", line 1003, in duplicated
    values = _ensure_data(values)
             ^^^^^^^^^^^^^^^^^^^^
  File "E:\pd_dev_0814\Lib\site-packages\pandas\core\algorithms.py", line 155, in _ensure_data
    return np.asarray(values).astype("uint8", copy=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: int() argument must be a string, a bytes-like object or a real number, not 'NAType'
>>>
```

### Issue Description

The error is not expected, the call to drop_duplicates should succeed.

### Expected Behavior

I expect this example to succeed. If the null value is removed, the call succeeds.

```
>>> dd = pd.DataFrame({"A": pd.Series([True, False], dtype="bool[pyarrow]")})
>>> dd.drop_duplicates()
       A
0   True
1  False
>>>
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit              : 6e194cf9eadedbf5506da5ba0939d700f46ba3ba
python              : 3.11.2.final.0
python-bits         : 64
OS                  : Windows
OS-release          : 10
Version             : 10.0.22621
machine             : AMD64
processor           : Intel64 Family 6 Model 85 Stepping 7, GenuineIntel
byteorder           : little
LC_ALL              : None
LANG                : None
LOCALE              : English_United States.1252

pandas              : 2.1.0rc0
numpy               : 1.25.2
pytz                : 2023.3
dateutil            : 2.8.2
setuptools          : 68.0.0
pip                 : 23.2.1
Cython              : None
pytest              : 7.0.1
hypothesis          : None
sphinx              : 4.2.0
blosc               : None
feather             : None
xlsxwriter          : None
lxml.etree          : None
html5lib            : None
pymysql             : None
psycopg2            : None
jinja2              : 3.1.2
IPython             : 8.14.0
pandas_datareader   : None
bs4                 : 4.12.2
bottleneck          : None
brotli              : None
dataframe-api-compat: None
fastparquet         : None
fsspec              : None
gcsfs               : None
matplotlib          : None
numba               : None
numexpr             : None
odfpy               : None
openpyxl            : None
pandas_gbq          : None
pyarrow             : 12.0.0
pyreadstat          : None
pyxlsb              : None
s3fs                : None
scipy               : None
snappy              : None
sqlalchemy          : None
tables              : None
tabulate            : 0.8.9
xarray              : None
xlrd                : None
zstandard           : None
tzdata              : 2023.3
qtpy                : None
pyqt5               : None
>>>

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #54667 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

cc @mroeschke 

More generally, how do we want to proceed with Arrow EA fixes after 2.1 is out?

### CODE REVIEW

File: pandas/tests/series/methods/test_drop_duplicates.py [hunk 1]
```diff
@@ -249,3 +249,10 @@ def test_drop_duplicates_ignore_index(self):
         result = ser.drop_duplicates(ignore_index=True)
         expected = Series([1, 2, 3])
         tm.assert_series_equal(result, expected)
+
+    def test_duplicated_arrow_dtype(keep):
```
Comment (mroeschke) [inline pandas/tests/series/methods/test_drop_duplicates.py:253]: Did you intend to parameterize over `keep`?

**Comment by mroeschke** [comment]
> More generally, how do we want to proceed with Arrow EA fixes after 2.1 is out?

Since Arrow support isn't new anymore I suppose we shouldn't be backporting bug fixes as aggressively anymore, but I would still be okay with it

File: pandas/tests/series/methods/test_drop_duplicates.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/series/methods/test_drop_duplicates.py]: Nope, that was a lesson that you shouldn't copy paste without double checking

**Comment by phofl** [comment]
Ok then maybe lets backport low risk fixes and decide on the others on a case by case basis

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**